### PR TITLE
Completed Validation specs n+1

### PR DESCRIPTION
**Pull request by AsimKhan2019**
For this assignment I @AsimKhan2019 and @benshidanny11 completed this assignment and task. For this assignment we created separate test conditions for the following scenarios: 

User's Index Page
User's Show Page
Post's Show Page
Posts' Index Page 

We covered most of the scenarios and test cases that there could be for this assignment, some of the test cases applied are:

See the list of all users 
See the list of all posts 
See the button to navigate to the list 
See the list of all the comments 
See the post details
See all the associated comments with their relevant post

For this assignment, we used the bullet gem to get instructions on where to include an option in the application. We added the included requirements in the controllers and models where it was necessary. 

To run these tests, we used the following commands: 

bundle exec rspec spec/views/users/index_html_erb_spec.rb
bundle exec rspec spec/views/users/show_html_erb_spec.rb
bundle exec rspec spec/views/posts/index_html_erb_spec.rb
bundle exec rspec spec/views/users/show_html_erb_spec.r

P.S. This was a very difficult assignment as there was no clarity on how to complete this assignment as we are running this application on Windows. We still require more guidance and peer-to-peer interaction to help us better understand how to do more for this assignment.

### CODE REVIEW

**Review by BenMukebo — COMMENTED** [review]
## STATUS: INVALID ❌ 

Hi  @AsimKhan2019, @benshidanny11,

I can't miss felicitating your efforts and energy, you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request 

- [ ] You have submitted a project, that doesn't have commits from all members in your pair programming group. Kindly note that this is a pair programming activity and all the members should commit to the project.
- Note the next reviewer will check the commit history to make sure that everybody has contributed code on the Pull Request on each project..

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@BenMukebo) in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **Does not count** into the code reviews limit.

File: spec/views/posts/index_html_erb_spec.rb [hunk 1]
```diff
@@ -0,0 +1,63 @@
+require 'rails_helper'
+
+RSpec.describe 'Posts', type: :system, js: true do
```
Comment (Lucas-Masaba) [inline spec/views/posts/index_html_erb_spec.rb:3]: [OPTIONAL]
- If you want to check if all the tests are passing on your terminal, you can remove `...type: :system, js: true` and replace all that with `type: :feature`. 👍

**Review by Lucas-Masaba — APPROVED** [review]
# Hello @AsimKhan2019 and @benshidanny11  👋 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

almost-there

## Highlights 🍾 
- Tests are passing ✔️ 
- Solved N + 1 problems 👍 
- Linter checks are passing ✔️ 

Despite mentioning that you found the task rather difficult, you still managed to succeed. This [cheatsheet](https://gist.github.com/zhengjia/428105) can help you out in the future.

## Optional suggestions 💭 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏 💻 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Lucas-Masaba in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by AsimKhan2019** [comment]
@Lucas-Masaba Thank you for the approval

File: spec/views/posts/index_html_erb_spec.rb (hunk 1 above)
Comment (AsimKhan2019) [reply spec/views/posts/index_html_erb_spec.rb:3]: @Lucas-Masaba Thank you, I will update the code according to this.

**Pull request merged**